### PR TITLE
fix engine configuration override

### PR DESCRIPTION
**Pull request by AvivCx**
### Description

Fixing scan configuration override

Expected behavior:
when adding 'engineConfiguration' value to config-as-code file, CxFlow should override global configuration and use value from config file

### References

Fix: #549
see Config as code [wiki page](https://github.com/checkmarx-ltd/cx-flow/wiki/Config-As-Code)

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentation is a Wiki Update, please indicate desired changes within PR MD Comment*
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

File: src/main/java/com/checkmarx/flow/sastscanning/ScanRequestConverter.java [hunk 1]
```diff
@@ -48,15 +48,23 @@ public CxScanParams toScanParams(ScanRequest scanRequest) throws CheckmarxExcept
...
-                log.debug("Using scan configuration ID: {}, name: '{}'.", configId, configName);
-                scanRequest.setScanConfiguration(configName);
-            } else {
-                log.warn("Unable to retrieve scan settings for the existing project (ID {}).", projectId);
+
+            if (!ScanUtils.empty(scanRequest.getScanConfiguration())){
+                String scanConfiguration = scanRequest.getScanConfiguration();
+                log.debug("overriding scan configuration with '{}'", scanConfiguration);
+                scanRequest.setScanConfiguration(scanConfiguration);
+            }
```
Comment (NimrodGolan) [inline src/main/java/com/checkmarx/flow/sastscanning/ScanRequestConverter.java:56]: Something won't make sense to me - if scan request does has the scan configuration value, why setting it again inside scan request?

File: src/main/java/com/checkmarx/flow/sastscanning/ScanRequestConverter.java [hunk 2]
```diff
@@ -48,15 +48,23 @@ public CxScanParams toScanParams(ScanRequest scanRequest) throws CheckmarxExcept
...
-                String configName = scannerClient.getScanConfigurationName(configId);
-                log.debug("Using scan configuration ID: {}, name: '{}'.", configId, configName);
-                scanRequest.setScanConfiguration(configName);
-            } else {
-                log.warn("Unable to retrieve scan settings for the existing project (ID {}).", projectId);
+
+            if (!ScanUtils.empty(scanRequest.getScanConfiguration())){
+                String scanConfiguration = scanRequest.getScanConfiguration();
+                log.debug("overriding scan configuration with '{}'", scanConfiguration);
+                scanRequest.setScanConfiguration(scanConfiguration);
```
Comment (alex-ko-dev) [inline src/main/java/com/checkmarx/flow/sastscanning/ScanRequestConverter.java:55]: Right, this looks like assigning scanConfiguration to itself.
Also, it seems that all the needed changes are already done in ConfigurationOverrider, so perhaps changes in ScanRequestConverter are not required.

File: src/main/java/com/checkmarx/flow/sastscanning/ScanRequestConverter.java (hunk 1 above)
Comment (AvivCx) [reply src/main/java/com/checkmarx/flow/sastscanning/ScanRequestConverter.java]: Fixed

File: src/main/java/com/checkmarx/flow/sastscanning/ScanRequestConverter.java (hunk 2 above)
Comment (AvivCx) [reply src/main/java/com/checkmarx/flow/sastscanning/ScanRequestConverter.java]: before my changes, CxFlow didn't check the configuration value in the scan request (that was overriden in the configuration-overrider)

**Review by NimrodGolan — APPROVED** [review]

**Pull request merged**